sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -950,6 +951,84 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
       checkAnswer(spark.table("t2"), Nil)
     }
   }
+
+  test("SPARK-35106: insert overwrite with custom partition path") {
```

Comment:
does this test fail before this PR? and do we need to test dynamic partition overwrite?

Reply:
With the change in DebugFileSystem, this test fails with
```
== Results ==
!== Correct Answer - 1 ==   == Spark Answer - 0 ==
struct<>                   struct<>
![2,1,1]
```
Will add a test for dynamic partition overwriting to custom partition path.